### DISCUSSION TITLE
Is the C Standard C90 or C11?

### DISCUSSION THREAD

**Discussion by zsw007** [libuv/libuv#4134] [Q&A] [answered]
Hi, I'm wondering if some clarification could be provided as to whether libuv is a C90 or C11 project.

CMakeLists.txt seems to set the standard as C90:

https://github.com/libuv/libuv/blob/0d78f3c758fdc41019093353a7ff4fed83005ac9/CMakeLists.txt#L23

However, stdatomic.h is being used, which is C11:

https://github.com/libuv/libuv/blob/0d78f3c758fdc41019093353a7ff4fed83005ac9/src/unix/async.c#L29

From discussions in issues https://github.com/libuv/libuv/issues/3714 https://github.com/libuv/libuv/issues/3887 https://github.com/libuv/libuv/pull/3950, it seems that the intention is to continue using stdatomic.h.

However, by setting `CMAKE_C_STANDARD` to 90, the `-std=gnu90` flag gets used when building with the clang compiler on z/OS, which explicitly disables features that are not C90. I'm not sure if this issue has popped up on other platforms, but it appears to be a [documented behavior of CMake](https://cmake.org/cmake/help/latest/prop_tgt/C_STANDARD.html). If libuv is now using C11 features, this behavior doesn't seem to be intended. Should the C Standard be updated to 11 in CMakeLists.txt, or is it being kept at 90 for other reasons?

**Comment by vtjnash** [comment, marked as answer]
The intent is to require C11, but with C90 syntax (except for `_Atomic` syntax, which does require C11). We should probably update the CMakeLists, since it is necessary to require C11 features (for `_Atomic`)
